google-common/src/main/scala/org/apache/pekko/stream/connectors/google/auth/AccessTokenCredentials.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, which was derived from Akka.
+ */
+
+/*
+ * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package org.apache.pekko.stream.connectors.google.auth
+
+import org.apache.pekko
+import pekko.annotation.InternalApi
+import pekko.http.scaladsl.model.headers.OAuth2BearerToken
+import pekko.stream.connectors.google.RequestSettings
+import com.google.auth.{ Credentials => GoogleCredentials }
+import com.typesafe.config.Config
+
+import java.net.URI
+import java.util
+import scala.concurrent.{ ExecutionContext, Future }
+
+@InternalApi
+private[auth] object AccessTokenCredentials {
+  def apply(c: Config): AccessTokenCredentials = AccessTokenCredentials(c.getString("project-id"), c.getString("token"))
```

Comment:
we may need a java api with create method

Reply:
This is private, its not meant to be created by java users and its only being used internally within Scala API (no other auth data classes have a java create method either as they are all private)